MoviesTVSentiments/app/src/main/java/com/google/moviestvsentiments/service/account/AccountSyncWorker.java

```diff
@@ -0,0 +1,32 @@
+package com.google.moviestvsentiments.service.account;
+
+import android.content.Context;
+import androidx.annotation.NonNull;
+import androidx.hilt.Assisted;
+import androidx.hilt.work.WorkerInject;
+import androidx.work.Worker;
+import androidx.work.WorkerParameters;
+
+/**
+ * A Worker that syncs pending Accounts with the server.
+ */
+public class AccountSyncWorker extends Worker {
+
+    private final AccountRepository repository;
+
+    @WorkerInject
+    public AccountSyncWorker(@Assisted @NonNull Context context,
+                             @Assisted @NonNull WorkerParameters parameters,
+                             AccountRepository repository) {
+        super(context, parameters);
+        this.repository = repository;
+    }
+
+    @Override
+    public Result doWork() {
+        if (repository.syncPendingAccounts()) {
+            return Result.success();
+        }
+        return Result.retry();
```

Comment:
Given the work is already periodic. I wonder if we need retry here or just failure() and wait for next time?

Looking into BackoffPolicy and setBackoffCriteria, I did not see they limit the number of retries. So a small concern is the app retry endlessly on some error that will never get fixed with retry.

Reply:
Is it possible to use server response http status code to determine whether to retry or not?

Reply:
The return value from the repository is based on the server response code. Do you mean retry if it is a 500 error and fail if it is a 400 error?

Reply:
There is a way for the Worker to get the number of attempts, so that could be used to return failure() instead of retry() once it reaches some max.

Reply:
Yes, I meant the detailed code. Generally can retry for 500 and fail for 400. I think limit the max retry is fine here. Thansk!